pkg/workspace/controllers/workspace_controller.go

```diff
@@ -463,6 +482,15 @@ func (c *WorkspaceReconciler) getAllQualifiedNodes(ctx context.Context, wObj *ka
 		}
 	}
 
+	// After looping, check for missing preferred nodes if feature gate is enabled
+	if featuregates.FeatureGates[consts.FeatureFlagDisableNodeAutoProvisioning] {
+		// Since we remove a preferredNode from the preferredNodeSet if we've seen it in the list of nodes matching the label,
+		// anything node left in the set means it does not match the label selector and we can surface an error.
+		if len(deletingPreferredNodes)+len(notReadyNodes)+preferredNodeSet.Len() > 0 {
+			return nil, fmt.Errorf("when node auto-provisioning is disabled, all preferred nodes must be ready, running, and match the label selector. The following nodes do not meet the required conditions: deleting nodes: %+v, not ready nodes: %+v, nodes missing label: %+v", deletingPreferredNodes, notReadyNodes, preferredNodeSet.UnsortedList())
```

Comment:
nit: preferredNodeSet could contain both nodes with no labels or nodes that don't exist at all

```suggestion
			return nil, fmt.Errorf("when node auto-provisioning is disabled, all preferred nodes must be ready, running, and match the label selector. The following nodes do not meet the required conditions: deleting nodes: %+v, not ready nodes: %+v, nodes missing label or do not exist: %+v", deletingPreferredNodes, notReadyNodes, preferredNodeSet.UnsortedList())
```